tests/unit_tests/test_optional_part.py

```diff
@@ -0,0 +1,191 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.item import Item
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+'''Tests for Optional Part: '''
+
+'''item.py add_ons'''
+
+#@pytest.mark.skip
+def test_item_year_incorrect_parameter_passed_raises_ValueError():
+    with pytest.raises(ValueError):
+        item = Item(year='year')
+        
+#@pytest.mark.skip        
+def test_int_valued_optional_parameters_accept_string_convertable_to_int():
+    item = Item(year='2014')
+    
+    assert isinstance(item.year, int)
+    assert item.year == 2014
+
+#@pytest.mark.skip
+def test_item_year_default_is_none():
+    item = Item()
+    
+    assert item.year is None
+    
+#@pytest.mark.skip
+def test_item_use_custom_year_if_passed():
+    item = Item(1989)
+    
+    assert item.year == 1989
+    
+#@pytest.mark.skip
+def test_item_instances_have_different_year_if_passed():
+    item1 = Item(1968)
+    item2 = Item(2017)
+    
+    assert item1.year == 1968
+    assert item2.year == 2017
+    
+#@pytest.mark.skip
+def test_item_subclass_instances_have_different_year_if_passed():
+    toaster = Electronics(2021)
+    hoodie = Clothing(2023)
+    
+    assert toaster.year == 2021
+    assert hoodie.year == 2023
+    
+#@pytest.mark.skip   
+def test_calculate_age_with_incorrect_arguments_raises_valueerror():
+    item = Item()
+    
+    with pytest.raises(TypeError):
+        item.calculate_age('unknown argument')
+        
+#@pytest.mark.skip         
+def test_calculate_age_returns_how_many_years_ago_from_today_item_was_created():
+    '''WARNING: Its date sensitive!!! 
+    Before testing make sure to change the value of 'current_year' for the actual current year.'''
+    current_year = 2023 
+    item = Item(year=2013)
+    # (current_year - item.year (2023-2013) = 10)
+    expected = 10
+    
+    actual = item.calculate_age()
```

Comment:
`calculate_age()` is a nice way of handling the problem of "if I store an age then it eventually becomes incorrect". Storing year then making age a calculation based on the current date is a very good idea.

You also ran into the hardest part of handling age calculations like this: making the tests pin the current year. The problem you ran into is that your tests don't have control over what `date.today()` returns. In general, you have to start mocking libraries in order to inject the right date time in your arrange step. A quick web search turned up SO answers that led to these python docs about partial mocking: https://docs.python.org/3/library/unittest.mock-examples.html#partial-mocking

This isn't the last time you'll run into mocking, and probably not the last time you'll run into the need to mock a `today()` function either–it'll come up whenever you're dealing with date or time calculations in any programming language. It can be kind of complicated to get it to work right, but it's basically always going to look like this. 😅